baseplate/lib/metrics.py

```diff
@@ -302,6 +303,9 @@ def send(self, elapsed: float) -> None:
 
         """
         serialized = self.name + (f":{(elapsed * 1000.0):g}|ms".encode())
+        if self.sample_rate < 1.0:
+            sampling_info = f"@{self.sample_rate:g}".encode()
+            serialized = serialized + b"|" + sampling_info
```

Comment:
worth benchmarking to be sure, but it might be more performant to do this the same way `Counter.send` works with building up a list of byte strings and then joining them at the end. that's generally one of the faster ways of doing this sort of string concatenation in python.